Source/WTF/wtf/text/WTFString.cpp

```diff
@@ -690,6 +691,42 @@ CString String::latin1() const
 }
 
 #if OS(MORPHOS) || OS(AMIGAOS)
+#if OS(AMIGAOS)
+ULONG GetLength(APTR str, LONG bytes, ULONG mib)
+{
+    return ICodesets->CodesetsStrLen((CONST_STRPTR) str,
```

Comment:
- When reading the MorphOS SDK this function says that a value of "NULL" for "string" returns -1. Can we optimize this by just returning -1 in that case?
- When the value "0" is used for "bytes" the guide says that this function immediately returns "0".

Reply:
> When reading the MorphOS SDK this function says that a value of "NULL" for "string" returns -1. Can we optimize this by just returning -1 in that case?

I am not sure if this is the case, because the function return ULONG, which if I am not wrong, it is an unsigned long.

> We also need "ConvertTagList" because "GetLength" internally uses this function. I can see that.

I created that method. If you mean the MorphOS GetLength(), then this doesn't apply here, because we created our own GetLength().

> Can't we make these functions static to the translation unit?

Can you please elaborate on what you mean here? I don't think I understand what you propose to do.

Reply:
===
> I am not sure if this is the case, because the function return ULONG, which if I am not wrong, it is an unsigned long.
I am reading the MorphOS SDK documentation. The "SYNOPSIS" for **GetLength()** in _charsets_ (version 51.3)
`LONG GetLength(APTR string, LONG bytes, ULONG mib);`
It also says, about the inputs:
> string - pointer to source string. NULL is safe, the result is -1.
and,
> bytes - number of valid bytes in the buffer. The special value -1 means
>      unlimited buffer, string is read up to the string terminator. Passing 0
>      here causes function to return immediately with result 0.
===
> I created that method. If you mean the MorphOS GetLength(), then this doesn't apply here, because we created our own GetLength().
Yeah, that is fine. I was just making an observation.
===
> Can you please elaborate on what you mean here? I don't think I understand what you propose to do.
I am just saying that where I generally can I try to restrict the visibility of functions. Turning these methods into static gives them static scope which means only that .c file has access to them. It is not really a bid deal.

Reply:
The only things to discuss is the first section of my comment above because perhaps we are looking at different SDK guides?